core/src/test/scala/unit/kafka/admin/ConsumerGroupServiceTest.scala

```diff
@@ -0,0 +1,133 @@
+    val topicArgs = Seq("--topic", s"${topics.head}:${(0 until numPartitions).mkString(",")}") ++
+      topicsWithoutPartitionsSpecified.flatMap { topic => Seq("--topic", topic) }
+    val groupService = consumerGroupService((args ++ topicArgs).toArray)
+
+    expect(admin.describeConsumerGroups(EasyMock.eq(Collections.singletonList(group)), anyObject()))
+      .andReturn(describeGroupsResult(ConsumerGroupState.DEAD))
+      .once()
+    expect(admin.describeTopics(EasyMock.eq(topicsWithoutPartitionsSpecified.asJava), anyObject()))
+      .andReturn(describeTopicsResult)
+      .once()
+    expect(admin.listOffsets(anyObject(), anyObject()))
+      .andAnswer(() => listOffsetsResult(getCurrentArguments.apply(0).asInstanceOf[util.Map[TopicPartition, OffsetSpec]]))
+      .once()
+    replay(admin)
+
+    val resetResult = groupService.resetOffsets()
+    assertEquals(Set(group), resetResult.keySet)
+    assertEquals(topicPartitions.toSet, resetResult(group).keySet)
+  }
+
+  private def consumerGroupService(args: Array[String]): ConsumerGroupService = {
+    new ConsumerGroupService(new ConsumerGroupCommandOptions(args)) {
+      override protected def createAdminClient(configOverrides: collection.Map[String, String]): Admin = {
+        admin
+      }
+    }
+  }
+
+  private def describeGroupsResult(groupState: ConsumerGroupState): DescribeConsumerGroupsResult = {
+    val member1 = new MemberDescription("member1", Optional.of("instance1"), "client1", "host1", null)
+    val description = new ConsumerGroupDescription(group,
+      true,
+      Collections.singleton(member1),
+      classOf[RangeAssignor].getName,
+      groupState,
+      new Node(1, "localhost", 9092))
+    new DescribeConsumerGroupsResult(Collections.singletonMap(group, KafkaFuture.completedFuture(description)))
+  }
+
+  private def listGroupOffsetsResult: ListConsumerGroupOffsetsResult = {
+    val offsets = topicPartitions.map { tp => tp -> new OffsetAndMetadata(100) }.toMap.asJava
+    val result: ListConsumerGroupOffsetsResult = mock(classOf[ListConsumerGroupOffsetsResult])
+    expect(result.partitionsToOffsetAndMetadata()).andReturn(KafkaFuture.completedFuture(offsets)).anyTimes()
+    replay(result)
+    result
+  }
+
+  private def listOffsetsResult(partitions: util.Map[TopicPartition, OffsetSpec]): ListOffsetsResult = {
+    assertEquals(topicPartitions.toSet, partitions.keySet.asScala)
+    assertEquals(classOf[OffsetSpec.LatestSpec], partitions.values.iterator.next.getClass)
+    val resultInfo = new ListOffsetsResult.ListOffsetsResultInfo(100, System.currentTimeMillis, Optional.of(1))
+    val futures = topicPartitions.map { tp => tp -> KafkaFuture.completedFuture(resultInfo) }.toMap
+    new ListOffsetsResult(futures.asJava)
+  }
+
+  private def describeTopicsResult: DescribeTopicsResult = {
+    val result: DescribeTopicsResult = mock(classOf[DescribeTopicsResult])
+    val topicDescriptions = topics.map { topic =>
```

Comment:
We almost create whole response. Also, we define `anyTimes` for the mock object. Hence, maybe we can replace the mock object by true object. The benefit is that it won't cause error if we call other methods of `DescribeTopicsResult` in the future.

Reply:
Was using mock since `DescribeTopicsResult` doesn't have a public constructor (and can't update the class since it is public API). Added a comment.

Reply:
You can add a test method that does this by having it in the right package and using a package private constructor.